src/lib/autoinstall/clients/autoyast.rb

```diff
@@ -0,0 +1,196 @@
+
+        Yast.include self, "autoinstall/dialogs.rb"
+        Yast.include self, "autoinstall/conftree.rb"
+      end
+
+      def main
+        if Yast::Y2ModuleConfig.GroupMap.empty?
+          Yast::Wizard.CreateDialog
+          Yast::Popup.Error(_("Error while reading configuration data."))
+          Yast::Wizard.CloseDialog
+          return :abort
+        end
+
+        turn_off_signature_checks
+
+        @cmdline = {
+          "id"         => "autoyast",
+          "help"       => _("AutoYaST"),
+          "guihandler" => fun_ref(method(:auto_sequence), "any ()"),
+          "actions"    => {
+            "file"         => {
+              "handler" => fun_ref(
+                method(:run_ui),
+                "boolean (map <string, string>)"
+              ),
+              "help"    => "file operations"
+            },
+            "module"       => {
+              "handler" => fun_ref(
+                method(:run_ui),
+                "boolean (map <string, string>)"
+              ),
+              "help"    => "module specific operations"
+            },
+            "list-modules" => {
+              "handler" => fun_ref(
+                method(:list_modules),
+                "void ()"
+              ),
+              "help"    => "list of known modules"
+            }
+          },
+          "options"    => {
+            "filename" => { "type" => "string", "help" => "filename=XML_PROFILE" },
+            "modname"  => { "type" => "string", "help" => "modname=AYAST_MODULE" }
+          },
+          "mappings"   => {
+            "file"   => ["filename", "modname"],
+            "module" => ["filename", "modname"]
+          }
+        }
+
+        ret = Yast::CommandLine.Run(@cmdline)
+        log.debug("ret = #{ret}")
+
+        Yast::AddOnProduct.CleanModeConfigSources
+        :exit
+      end
+
+      # Run the main UI
+      #
+      # @param options [Hash] Command line options
+      # @return true
+      def run_ui(options)
+        import_profile(options["filename"]) if options["filename"]
+        Yast::AutoinstConfig.runModule = options["modname"] || ""
+        auto_sequence
+        true
+      end
+
+      # Displays the list of known modules
+      #
+      # @param _options [Hash] Command line options
+      def list_modules(_options)
+        items = Yast::Y2ModuleConfig.ModuleMap.reduce([]) do |all, (name, mod)|
+          all << Item(name, mod["Name"])
+        end
+        Yast::CommandLine.PrintTable(
+          Header(_("Module"), _("Description")),
+          items
+        )
+      end
+
+    private
+
+      # Reads and imports a profile
+      #
+      # @param filename [String] Profile path
+      def import_profile(filename)
+        if !Yast::Profile.ReadXML(filename)
+          Yast::Popup.Error(
+            _(
+              "Error while parsing the control file.\n" \
+                "Check the log files for more details or fix the\n" \
+                "control file and try again.\n"
```

Comment:
is it really control file or autoyast profile?

Reply:
You are right, it is the AutoYaST profile. The problem that even in the AutoYaST documentation, sometimes we refer to the AutoYaST profile as the "control file". I will fix.